### ISSUE TITLE
[oatpp] update to 1.2.5

### ISSUE THREAD

**Issue by liujin001** [microsoft/vcpkg#17599]
Library name: oatpp

New version number: 1.2.5

Release notes can be found at https://github.com/oatpp/oatpp/releases/tag/1.2.5

**Comment by JonLiu1993** [comment]
@liujin001 ,Thanks for your pr,according to the error display of CI, the components port of oatpp also needs to be upgraded along with oatpp. could you upgrade all components together in one pr?

**Comment by xanderdunn** [comment]
@autoantwort @JonLiu1993 When I attempt to link against liboatpp.a 1.2.5 built with vcpkg it fails with the following error:
```
/usr/bin/ld: /home/xander/dev/vcpkg/installed/x64-linux/debug/lib/oatpp-1.2.5/liboatpp.a(Environment.cpp.o): relocation R_X86_64_TPOFF32 against symbol `_ZN5oatpp4base11Environment25m_threadLocalObjectsCountE' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /home/xander/dev/vcpkg/installed/x64-linux/debug/lib/oatpp-1.2.5/liboatpp.a(MemoryPool.cpp.o): relocation R_X86_64_TPOFF32 against `_ZGVZN5oatpp4base6memory27ThreadDistributedMemoryPool6obtainEvE5index' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /home/xander/dev/vcpkg/installed/x64-linux/debug/lib/oatpp-1.2.5/liboatpp.a(ChunkedBuffer.cpp.o): relocation R_X86_64_TPOFF32 against symbol `_ZGVZN5oatpp10collection10LinkedListISt10shared_ptrINS_4data6stream13ChunkedBuffer5ChunkEEE14LinkedListNode20LinkedList_Node_Pool7getPoolEvE4pool' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: CMakeFiles/my_shared_lib.dir/src/Global.cpp.o(.text._ZZSt9call_onceIZN5quill6detail13BackendWorker3runEvEUlvE_JEEvRSt9once_flagOT_DpOT0_ENKUlvE0_clEv[_ZZSt9call_onceIZN5quill6detail13BackendWorker3runEvEUlvE_JEEvRSt9once_flagOT_DpOT0_ENKUlvE0_clEv]+0x1a): unresolvable R_X86_64_DTPOFF32 relocation against symbol `_ZSt15__once_callable@@GLIBCXX_3.4.11'
/usr/bin/ld: final link failed: Nonrepresentable section on output
collect2: error: ld returned 1 exit status
```

"can not be used when making a shared object; recompile with -fPIC"

Does -fPIC need to be added to the port's compiler flags so that it's suitable for inclusion in shared libraries?

**Comment by JonLiu1993** [comment]
> @autoantwort @JonLiu1993 When I attempt to link against liboatpp.a 1.2.5 built with vcpkg it fails with the following error:
> 
> ```
> /usr/bin/ld: /home/xander/dev/vcpkg/installed/x64-linux/debug/lib/oatpp-1.2.5/liboatpp.a(Environment.cpp.o): relocation R_X86_64_TPOFF32 against symbol `_ZN5oatpp4base11Environment25m_threadLocalObjectsCountE' can not be used when making a shared object; recompile with -fPIC
> /usr/bin/ld: /home/xander/dev/vcpkg/installed/x64-linux/debug/lib/oatpp-1.2.5/liboatpp.a(MemoryPool.cpp.o): relocation R_X86_64_TPOFF32 against `_ZGVZN5oatpp4base6memory27ThreadDistributedMemoryPool6obtainEvE5index' can not be used when making a shared object; recompile with -fPIC
> /usr/bin/ld: /home/xander/dev/vcpkg/installed/x64-linux/debug/lib/oatpp-1.2.5/liboatpp.a(ChunkedBuffer.cpp.o): relocation R_X86_64_TPOFF32 against symbol `_ZGVZN5oatpp10collection10LinkedListISt10shared_ptrINS_4data6stream13ChunkedBuffer5ChunkEEE14LinkedListNode20LinkedList_Node_Pool7getPoolEvE4pool' can not be used when making a shared object; recompile with -fPIC
> /usr/bin/ld: CMakeFiles/my_shared_lib.dir/src/Global.cpp.o(.text._ZZSt9call_onceIZN5quill6detail13BackendWorker3runEvEUlvE_JEEvRSt9once_flagOT_DpOT0_ENKUlvE0_clEv[_ZZSt9call_onceIZN5quill6detail13BackendWorker3runEvEUlvE_JEEvRSt9once_flagOT_DpOT0_ENKUlvE0_clEv]+0x1a): unresolvable R_X86_64_DTPOFF32 relocation against symbol `_ZSt15__once_callable@@GLIBCXX_3.4.11'
> /usr/bin/ld: final link failed: Nonrepresentable section on output
> collect2: error: ld returned 1 exit status
> ```
> 
> "can not be used when making a shared object; recompile with -fPIC"
> 
> Does -fPIC need to be added to the port's compiler flags so that it's suitable for inclusion in shared libraries?

@xanderdunn ,Have you tried adding -fPIC to the port's compiler flag? I investigated your idea is feasible

**Comment by xanderdunn** [comment]
I wasn't specific about my usage in my above post. When I'm linking an executable in CMAKE against oatpp it works, but here I'm trying to link a shared library against oatpp and thus I see the above error.

Thanks @JonLiu1993. I added this to ports/oatpp/portfile.cmake:
```
set(CMAKE_CXX_FLAGS "-fPIC")
```

Then I uninstalled and reinstalled:
```
$ vcpkg remove oatpp --recurse
$ vcpkg build oatpp
$ vcpkg install oatpp
```
This appears to have rebuilt oatpp, but I still get the error when linking a shared library that links to oatpp.

This is the first time I'm modifying a vcpkg port, as well as the first time I'm making a C++ shared library, so I'm not experienced with how this ought to be done.
